### PR TITLE
Deprecate certificate pending

### LINKED ISSUE #1925: Signer computes what to sign on its own
## Why
The signer needs to compute the next message to sign independently and without relying on the pending certificate of the aggregator.

## What
Replace the usage of the Pending Certificate in the signer with its own computation of which is the next message to sign.

## How

- [x] Adapt the mechanism of the aggregator to compute the next signed entity type to sign
- [x] Store the signed entity type signed persistently when it has been successfully (when signature is sent)
- [x] Prune or keep last 2 epochs in the store?
- [x] Remove all pending certificate code in the signer
- [x] Deprecate the `/pending-certificate` route in OpenAPI

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR deprecate the `CertificatePendingMessage` both in rust code and in the openapi. It also remove the remaining certificate pending code in `mithril-signer`.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Comments

This is a follow up on #1970, so it must be merged after it.

## Issue(s)

Closes #1925

### CODE REVIEW

File: mithril-common/src/messages/certificate_pending.rs [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 use serde::{Deserialize, Serialize};
 
 use crate::entities::{CardanoDbBeacon, Epoch, ProtocolParameters, SignedEntityType};
 use crate::messages::SignerMessagePart;
 
 /// Structure to transport [crate::entities::CertificatePending] data.
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[deprecated(
+    since = "0.4.61",
+    note = "Exist only for backward-compatibility, will be removed in the future"
```
Comment (jpraynaud) [inline mithril-common/src/messages/certificate_pending.rs:11]:
```suggestion
    note = "Exists only for backward-compatibility, will be removed in the future"
```

**Review by jpraynaud — APPROVED** [review]
LGTM

File: mithril-signer/src/services/aggregator_client.rs [hunk 2]
```diff
@@ -214,36 +205,6 @@ impl AggregatorClient for AggregatorHTTPClient {
         }
     }
```
Comment (sfauvel) [inline mithril-signer/src/services/aggregator_client.rs:207]: The /certificate_pending still exist in openapi.yaml even if it was deprecated. 
Is it normal to remove all tests ? What happen when the route is called ?

**Review by sfauvel — APPROVED** [review]
LGTM

**Review by dlachaume — APPROVED** [review]

File: mithril-common/src/test_utils/apispec.rs [hunk 3]
```diff
@@ -433,12 +443,12 @@ components:
     fn test_validate_ok_when_request_without_body_and_expects_response() {
         APISpec::from_file(&APISpec::get_default_spec_file())
             .method(Method::GET.as_str())
-            .path("/certificate-pending")
+            .path("/")
```
[the remark this answers is not in the document]
Comment (Alenar) [reply mithril-common/src/test_utils/apispec.rs:446]: Agree with that :+1: , as you say we should do a refactoring PR to solve that problem.

**Pull request merged**